src/app/core/services/http.service.ts

```diff
@@ -0,0 +1,348 @@
+    return this.http
+      .get<SignUpResponse>(this.url + this.usersPath + '/' + userId)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public updateUser(
+    userId: string,
+    params: SignUpBody
+  ): Observable<SignUpResponse | Observable<never>> {
+    return this.http
+      .put<SignUpResponse>(this.url + this.usersPath + '/' + userId, params)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public deleteUser(
+    userId: string
+  ): Observable<SignUpResponse | Observable<never>> {
+    return this.http
+      .delete<SignUpResponse>(this.url + this.usersPath + '/' + userId)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public signUp(
+    params: SignUpBody
+  ): Observable<SignUpResponse | Observable<never>> {
+    return this.http
+      .post<SignUpResponse>(this.url + this.signUpPath, params)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public singIn(
+    params: SignInBody
+  ): Observable<SignInResponseBody | Observable<never>> {
+    return this.http
+      .post<SignInResponseBody>(this.url + this.singInPath, params)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+  public createBoard(params: BoardBody): Observable<Board | Observable<never>> {
+    return this.http
+      .post<Board>(this.url + this.boardsPath, params)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+  public getAllBoards(): Observable<Board[] | Observable<never>> {
+    return this.http
+      .get<Board[]>(this.url + this.boardsPath)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public getBoard(boardId: string): Observable<Board | Observable<never>> {
+    return this.http
+      .get<Board>(this.url + this.boardsPath + '/' + boardId)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+  public updateBoard(
+    boardId: string,
+    params: BoardBody
+  ): Observable<Board | Observable<never>> {
+    return this.http
+      .put<Board>(this.url + this.boardsPath + '/' + boardId, params)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+  public deleteBoard(boardId: string): Observable<Board | Observable<never>> {
+    return this.http
+      .delete<Board>(this.url + this.boardsPath + '/' + boardId)
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+  public getAllColumns(
+    boardId: string
+  ): Observable<Column[] | Observable<never>> {
+    return this.http
+      .get<Column[]>(
+        this.url + this.boardsPath + '/' + boardId + this.columnsPath
+      )
+      .pipe(catchError(async (err) => this.httpError.catchErrors(err)));
+  }
+
+  public getColumn(
+    boardId: string,
+    columnId: string
+  ): Observable<Column | Observable<never>> {
+    return this.http
+      .get<Column>(
+        this.url +
```

Comment:
Templates string will help you to make code more readable (https://javascript.info/string)